core/sail/shacl/src/test/java/org/eclipse/rdf4j/sail/shacl/OrderingTest.java

```diff
@@ -0,0 +1,185 @@
+ * Copyright (c) 2018 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+
+package org.eclipse.rdf4j.sail.shacl;
+
+import org.eclipse.rdf4j.IsolationLevels;
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+import org.eclipse.rdf4j.model.vocabulary.FOAF;
+import org.eclipse.rdf4j.model.vocabulary.RDF;
+import org.eclipse.rdf4j.model.vocabulary.RDFS;
+import org.eclipse.rdf4j.model.vocabulary.SHACL;
+import org.eclipse.rdf4j.repository.sail.SailRepository;
+import org.eclipse.rdf4j.repository.sail.SailRepositoryConnection;
+import org.eclipse.rdf4j.sail.SailConnection;
+import org.eclipse.rdf4j.sail.memory.MemoryStore;
+import org.eclipse.rdf4j.sail.shacl.AST.ShaclProperties;
+import org.eclipse.rdf4j.sail.shacl.mock.MockConsumePlanNode;
+import org.eclipse.rdf4j.sail.shacl.planNodes.InnerJoin;
+import org.eclipse.rdf4j.sail.shacl.planNodes.PlanNode;
+import org.eclipse.rdf4j.sail.shacl.planNodes.Select;
+import org.eclipse.rdf4j.sail.shacl.planNodes.Sort;
+import org.eclipse.rdf4j.sail.shacl.planNodes.TrimTuple;
+import org.eclipse.rdf4j.sail.shacl.planNodes.Tuple;
+import org.eclipse.rdf4j.sail.shacl.planNodes.UnBufferedPlanNode;
+import org.eclipse.rdf4j.sail.shacl.planNodes.ValuesBackedNode;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+
+/**
+ * @author Håvard Ottestad
+ */
+public class OrderingTest {
+
+	private final static ValueFactory vf = SimpleValueFactory.getInstance();
+
+	@Test
+	public void testSelect() {
+		MemoryStore repository = new MemoryStore();
+		repository.init();
+
+		try (SailConnection connection = repository.getConnection()) {
+			connection.begin(IsolationLevels.NONE);
+			connection.addStatement(RDFS.RESOURCE, RDF.TYPE, RDFS.RESOURCE);
+			connection.addStatement(RDFS.CLASS, RDF.TYPE, RDFS.RESOURCE);
+			connection.addStatement(RDFS.SUBCLASSOF, RDF.TYPE, RDFS.RESOURCE);
+			connection.commit();
+
+			Select select = new Select(connection, "?a <" + RDF.TYPE + "> []", "*");
+			List<Tuple> tuples = new MockConsumePlanNode(select).asList();
+
+			String actual = Arrays.toString(tuples.toArray());
+
+			Collections.sort(tuples);
+
+			String expected = Arrays.toString(tuples.toArray());
+
+			assertEquals(expected, actual);
```

Comment:
While these tests (after reading back and forth a bit) seem sensible, I'm a bit lost on what this has to do with issue GH-1574. Did the fix you added change ordering behavior somehow?

Reply:
Ordering test is a renamed file where I’ve moved in some old tests and also added one that checks that the set used to store the target nodes is sorted. Which is an assumption used further down the line.